packages/@aws-cdk/aws-route53-targets/lib/cloudfront-target.ts

```diff
@@ -11,12 +12,36 @@ export class CloudFrontTarget implements route53.IAliasRecordTarget {
    */
   public static readonly CLOUDFRONT_ZONE_ID = 'Z2FDTNDATAQYW2';
 
+  /**
+   * Get the hosted zone id for the current scope.
+   *
+   * @param stack - Stack in which the resource is defined
+   */
+  public static getHostedZoneId(stack: Stack) {
+    if (!CloudFrontTarget.stackPartitionHostedZoneMapping.hasOwnProperty(stack.stackId)) {
+      CloudFrontTarget.stackPartitionHostedZoneMapping[stack.stackId] = new CfnMapping(stack, 'AWSCloudFrontPartitionHostedZoneIdMap', {
```

Comment:
is it advantageous to have a mapping instead of a condition?
using a condition would look something like [this](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-kinesis/lib/stream.ts#L298-L319)

Reply:
I think this is fine actually... after some discussion, we decided that since we expect future values (more than 2.. i.e. with the addition of GovCloud), it would be more meaningful as a map instead of evaluating conditions. 

resolving this comment.

Reply:
@shivlaks I think the usage of a mapping is preferable here to aid when CloudFront support eventually lands in GovCloud to avoid complex branching when there are more than two hosted zone ids to pick from.